### ISSUE TITLE
Add password field to VirtualMachine

### ISSUE THREAD

**Issue by ams0** [Azure/azure-service-operator#1167]
**Describe the bug**

Virtual Machine controller accepts Windows for `OSType` but won't allow anything but SSH key to be provided.

**To Reproduce**

Create a VM with a PlatformURN pointing to a Windows image (for example `MicrosoftWindowsServer:WindowsServer:2019-Datacenter:latest` , try to provide a password (will fail).

**Expected behavior**
Being able to provision a windows VM with password. Extra goal: being able to refer to a kubernetes secret instead of explicitly writing it in the YAML (this can open the possibility to get the password from, say, Azure Keyvault).

**Additional context**

We are scoping a project where a mix of Linux and Windows workloads are needed (the latter cannot be containerized as they involve GUI apps); being able to create and manage the lifecycle of VM's thru native k8s objects would be great.

**Comment by huangpf** [comment]
Hi @ams0 -- do you know how to set a secret value in the project's running environment? If you can manage to add one with the ```password``` key, the operator will use it as the password value. 

Reference: https://github.com/Azure/azure-service-operator/blob/master/pkg/resourcemanager/vm/reconcile.go#L48

```
adminPassword := string(secret["password"])
```

**Comment by cesar-younes** [comment]
Hello @huangpf I was trying to create a Windows VM. What I understood from Line 38 above the line you referenced is that it checks if a password exists and if it does not it creates one and stores it in a secret. When I tried creating the vm I saw that a secret was created under the same namespace as the azurevirtualmachine  but I got this error: 
```
compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="Required parameter 'adminPassword' is missing (null)." Target="adminPassword"
```

When explicitly specify adminPassword in the yaml of the azurevirtualmachine then I get this error:
```
The supplied password must be between 8-123 characters long and must satisfy at least 3 of password complexity requirements from the following:\r\n1) Contains an uppercase character\r\n2) Contains a lowercase character\r\n3) Contains a numeric digit\r\n4) Contains a special character\r\n5) Control characters are not allowed" Target="adminPassword"
```
Can you elaborate more on the solution you described?

**Comment by cesar-younes** [comment]
I deleted the azurevirtualmachine and made sure the secret was deleted then tried applying the same YAML file again. Now I got a different error which is strange because I didn't change anything. I'm guessing only but since the only thing different between the 2 tries was the password that was generated, there may be an issue with how the password is passed on to Azure by the operator or maybe by the Azure SDK for go.
